sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala

```diff
@@ -31,14 +31,22 @@ import org.apache.spark.sql.test.SQLTestUtils
 class HiveMetadataCacheSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
 
   test("SPARK-16337 temporary view refresh") {
-    withTempView("view_refresh") {
+    checkRefreshView(isTemp = true)
+  }
+
+  test("view refresh") {
```

Comment:
We didn't cover the persistent view case for refresh, that's why the bug happens...

Reply:
Yes. We need to ask contributors for adding more test cases when reviewing the PR.